charts/shoot-core/components/charts/konnectivity-agent/templates/daemonset.yaml

```diff
@@ -0,0 +1,69 @@
+{{- if semverCompare ">= 1.18" .Values.kubernetesVersion }}
+apiVersion: {{ include "daemonsetversion" . }}
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: konnectivity-agent
+    app: konnectivity-agent
+    garden.sapcloud.io/role: system-component
+    origin: gardener
+  namespace: kube-system
+  name: konnectivity-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: konnectivity-agent
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: konnectivity-agent
+        app: konnectivity-agent
+      annotations:
+        scheduler.alpha.kubernetes.io/critical-pod: ''
+{{- if .Values.podAnnotations }}
+{{ toYaml .Values.podAnnotations | indent 8 }}
+{{- end }}
+    spec:
+      hostNetwork: true
+      serviceAccountName: konnectivity-agent
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+      containers:
+        - name: konnectivity-agent
+          image: {{ index .Values.images "konnectivity-agent" }}
+          command:
+            - /proxy-agent
+          args:
+            - --logtostderr=true
+            - --ca-cert=/certs/konnectivity-agent/ca.crt
+            - --agent-cert=/certs/konnectivity-agent/konnectivity-agent.crt
+            - --agent-key=/certs/konnectivity-agent/konnectivity-agent.key
+            - --proxy-server-host={{ .Values.proxyHost }}
+            - --proxy-server-port=8132
```

Comment:
Use service account token volume projection to further increase the security and verify individual agents
- https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/e6e9418c08fd6619bccfdbd0345384fc1c554509/examples/kubernetes/konnectivity-agent.yaml#L46-L52
- https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/e6e9418c08fd6619bccfdbd0345384fc1c554509/examples/kubernetes/konnectivity-agent.yaml#L29

And on the server - https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/e6e9418c08fd6619bccfdbd0345384fc1c554509/examples/kubernetes/konnectivity-server.yaml#L30-L33
as this would use `TokenReview` API to validate the tokens.

Reply:
I remmember testing volume projection earlier and something didn't seem to work so I was skeptical a bit, I will try it again.

Reply:
It's enabled by default on gardener shoot clusters. For minikube you can enable it via:

```
minikube start --vm-driver=hyperkit \
  --extra-config=apiserver.service-account-signing-key-file=/var/lib/minikube/certs/sa.key \
  --extra-config=apiserver.service-account-issuer=https://foo.bar \
  --extra-config=apiserver.service-account-api-audiences=api
```

Reply:
tested locally after creating a token and it does not work `error: You must be logged in to the server (Unauthorized)`

Reply:
It seems that the `authentication-audience` is not working very well with the latest release tag at this point in time, plus it requires more migration steps (e.g., creating new kubeconfigs, restarting components attached, ..) which can be undesirable when we are first trying it out. The certificate option works well for now IMO, so lets keep it simple for now and approach this with a later release tag.

Reply:
What do you mean by that? This token cannot be used for K8S API server authentication - it's only used to verify that the caller is actually a Pod (or in this case the network proxy server to verify the identity of the network proxy agent):

```
kubectl create -f - -o yaml  << EOF
apiVersion: authentication.k8s.io/v1beta1
kind: TokenReview
spec:
  token: eyJhbGciOiJSUzI1NiIsImtpZ.....
  audiences:
  - test-audience
EOF
apiVersion: authentication.k8s.io/v1beta1
kind: TokenReview
spec:
  audiences:
  - sap
  token: eyJhbGciOiJSUzI1NiIsImtpZ.....
status:
  audiences:
  - test-audience
  authenticated: true
  user:
    extra:
      authentication.kubernetes.io/pod-name:
      - nginx
      authentication.kubernetes.io/pod-uid:
      - 1130f250-9291-4948-a25e-f9434306a596
    groups:
    - system:serviceaccounts
    - system:serviceaccounts:default
    - system:authenticated
    uid: ef2e048c-8681-419d-9ebd-a9224bef4395
    username: system:serviceaccount:default:default
```

Reply:
what I mean is you need to generate a kubeconfig and pass that as a flag for the konnectivity-server for the "system:konnectivity-server" uid to be attached to the token and for that the kubeconfig generation logic needs to be modified

Reply:
Okay although I didn't want to do this in this PR, I have refactored some of the code for token generation to get this to work correctly.